downloads/swan-lake-release-notes/swan-lake-alpha2/RELEASE_NOTE.md

```diff
@@ -2,41 +2,599 @@
+}
+
+function getBodyPartsAsStream(int arraySize = 8196) returns stream<byte[], io:Error>|mime:ParserError {
+
+    // Sample
+    byte[][] content = ["File Content".toBytes()];
+    stream<byte[], io:Error> byteStream = content.toStream();
+    mime:Entity entity = new;
+    entity.setByteStream(byteStream);
+    stream<byte[], io:Error>|mime:ParserError str = entity.getByteStream();
+}
+```
+
+#### WebSocket Module Improvements
+
+Introduced the Sync client. This is the primary client of the WebSocket module. This client is capable of reading and writing messages synchronously.
+
+**Reading and Writing Text Messages**
+
+```ballerina
+websocket:Client wsClient = check new ("ws://echo.websocket.org");
+var err = wsClient->writeTextMessage("Text message");
+string textResp = check wsClient->readTextMessage();
+```
+
+**Reading and Writing Binary Messages**
+
+```ballerina
+websocket:Client wsClient = check new ("ws://echo.websocket.org");
+var err = wsClient->writeBinaryMessage("Binary message".toBytes());
+byte[] byteResp = check wsClient->readBinaryMessage();
+```
+
+##### GraphQL Module Improvements
+
+Ballerina GraphQL listeners can now be configured using the same configurations as the listener configurations in `http:Listener`. Additionally, a GraphQL service can be secured by defining a `maxQueryDepth` as an annotation to restrict the depth of a query before execution.
+
+```ballerina
+import ballerina/graphql;
+
+graphql:ListenerConfiguration configs = {
+    // http listener configurations
+};
+listener graphql:Listener graphqlListener = new (9090, configs);
+
+@graphql:ServiceConfigurration {maxQueryDepth: 3}
+ Service /graphql on graphqlListener {
+    // Service definition
+}
+```
+
+##### WebSub Module Improvements
+
+Included functionality to the `websub:SubscriberService` to respond with user-defined custom payloads/header parameters in error scenarios.
+
+```ballerina
+import ballerina/websub;
+
+listener websub:Listener subscriberListener = new (9001);
+
+service /subscriber on subscriberListener {
+    remote function onSubscriptionValidationDenied(websub:SubscriptionDeniedError msg) returns websub:Acknowledgement? {
+        websub:Acknowledgement ack = {
+            headers: {
+                "Content-Encoding": "gzip"
+            },
+            body: {
+                "message": "Successfully processed request"
+            }
+        };
+        return ack;
+    }
+
+    remote function onSubscriptionVerification(websub:SubscriptionVerification msg) returns 
+            websub:SubscriptionVerificationSuccess|websub:SubscriptionVerificationError {
+        if (msg.hubTopic == "https://www.sample.topic") {
+            return error websub:SubscriptionVerificationError("Hub topic not supported", 
+                                                              headers = {"Content-Encoding": "gzip"}, 
+                                                              body = {"message": "Hub topic not supported"});
```

Comment:
Shouldn't this be the else condition's body? @shafreenAnfar

Reply:
@Bhashinee

Reply:
Tagging @ayeshLK

Reply:
@pubudu91 in `WebSub Hub` either it could block / allow subscription based on `topic`. Here the `Hub` is denying the subscription verification based on the given `topic`.

Reply:
Ack.